src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let customJoin = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this[0] === null) {
+      customJoin += '';
+    } else {
+      customJoin += this[0];
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null || this[i] === ' ') {
+        customJoin += String(separator) + '';
```

Comment:
```suggestion
        customJoin += String(separator);
```
customJoin and String(separator) == string anyway